falcon/status_codes.py

```diff
@@ -14,135 +14,196 @@
 
 """HTTP status line constants."""
 
+from typing import Final
+
 # 1xx - Informational
-HTTP_100 = HTTP_CONTINUE = '100 Continue'
-HTTP_101 = HTTP_SWITCHING_PROTOCOLS = '101 Switching Protocols'
-HTTP_102 = HTTP_PROCESSING = '102 Processing'
+HTTP_100: Final[str] = '100 Continue'
+HTTP_CONTINUE = HTTP_100
```

Comment:
I think all the reassign should also be typed with `: Final[str]` since the final qualifier does not transfer. This types check correctly
```py
from typing import Final

HTTP_100: Final[str] = '100 Continue'
HTTP_CONTINUE = HTTP_100
HTTP_CONTINUE = 'foo'
```
while if we add the final type we get an error:
```py
from typing import Final

HTTP_100: Final[str] = '100 Continue'
HTTP_CONTINUE: Final[str] = HTTP_100
HTTP_CONTINUE = 'foo'
```
`main.py:5: error: Cannot assign to final name "HTTP_CONTINUE"  [misc]`

Reply:
Changed